sljit_src/sljitLir.h

```diff
@@ -258,20 +258,32 @@ extern "C" {
 /* --------------------------------------------------------------------- */
 
 /* The following argument type definitions are used by sljit_emit_enter,
-   sljit_set_context, sljit_emit_call, and sljit_emit_icall functions.
+   sljit_emit_call, and sljit_emit_icall functions.
 
-   As for sljit_emit_call and sljit_emit_icall, the first integer argument
+   For sljit_emit_call and sljit_emit_icall, the first integer argument
    must be placed into SLJIT_R0, the second one into SLJIT_R1, and so on.
    Similarly the first floating point argument must be placed into SLJIT_FR0,
    the second one into SLJIT_FR1, and so on.
 
-   As for sljit_emit_enter, the integer arguments can be stored in scratch
-   or saved registers. The first integer argument without _R postfix is
-   stored in SLJIT_S0, the next one in SLJIT_S1, and so on. The integer
-   arguments with _R postfix are placed into scratch registers. The index
-   of the scratch register is the count of the previous integer arguments
-   starting from SLJIT_R0. The floating point arguments are always placed
-   into SLJIT_FR0, SLJIT_FR1, and so on.
+   For sljit_emit_enter, the integer arguments can be stored in scratch
+   or saved registers. Scratch registers are identified by a _R suffix.
+
+   If only saved registers are used, then the allocation mirrors what is
+   done for the "call" functions but using saved registers, meaning that
+   the first integer argument goes to SLJIT_S0, the second one goes into
+   SLJIT_S1, and so on.
+
+   If scratch registers are used, then the way the integer registers are
+   allocated changes so that SLJIT_S0 will be assigned ONLY after the first
+   non scratch register.
+
+   Furthermore, the index that will be used for the scratch register depends
+   on how many previous integer registers (scratch or saved) were allocated,
+   starting with SLJIT_R0. See below for some examples.
+
+   The floating point arguments always use scratch registers (but not the
+   _R suffix like in the integer registers) and must be placed into SLJIT_FR0,
+   SLJIT_FR1, just like in the "call" functions.
 
    Note: if a function is called by sljit_emit_call/sljit_emit_icall and
```

Comment:
I don't understand this "Note", specially the references to the index affecting the floating point register index somehow and that would seem to conflict with the fact that floating point registers should be always using `SLJIT_FR0 ..`

Reply:
It wants to explain that floating point arguments cannot be moved to saved registers, so the _R prefix is meaningless, and not used.

Thank you for improving the text.

Reply:
I was referring to the note in https://github.com/zherczeg/sljit/blame/47ed75d0bcbe77f3e82b89e6e27f6b535815b821/sljit_src/sljitLir.h#LL276-L279C47

Reply:
The note explains that if an argument is not moved to a saved register, the register assigned to the argument "keeps" its value. I.e., if you put an argument to R2 for the call, it is still in R2 after emit_enter, unless it is moved to a saved register (R2 is undefined in the latter case, just like the other registers).